src/fiber/context.cr

```diff
@@ -0,0 +1,55 @@
+class Fiber
+  # :nodoc:
+  #
+  # The arch-specific make/swapcontext assembly relies on the Context struct and
+  # expects the following layout. Avoid moving the struct properties as it would
+  # require to update all the make/swapcontext implementations.
+  @[Extern]
+  struct Context
+    property stack_top : Void*
+    property resumable : LibC::Long
+
+    def initialize(@stack_top = Pointer(Void).null)
+      @resumable = 0
```

Comment:
can you use an `Enum` here instead of magic int numbers, as in:
```cr
enum ResumableState : LibC::Long
  State1 = 0
  State2 = 1
  # etc..
end
```

Reply:
@bew see https://github.com/crystal-lang/crystal/pull/6990#discussion_r228274432

Reply:
Ah yes, I remember now, but isn't the enum binary represented as an integer (so it should be directly usable from ASM) ?